docs/src/LinearAlgebra/vinberg.md

```diff
@@ -0,0 +1,53 @@
+```@meta
+CurrentModule = Oscar
+DocTestSetup = Oscar.doctestsetup()
+```
+
+# Vinberg's algorithm
+
+ A Lorentzian lattice $L$ is an integral $\Z$ - lattice in hyperbolic space.
+ $L$ is called reflective if the set of fundamental roots $\~{R}(L)$ is finite.
+
+ See for example [Tur17](@cite) for the theory of Arithmetic Reflection Groups and Reflective Lorentzian Lattices.
+
+## Description 
+ The algorithm constructs a fundamental polyhedron $P$ for a Lorentzian lattice $L$ by computing its fundamental roots $r$.
+
+ Choose $v_0$ with $v_0^2 > 0$ as a point that $P$ should contain.
+
+ Let $Q$ be the corresponding gram matrix of $L$. A fundamental root $r$ satisfies
+ - the vector $r$ is primitive
+ - reflection by $r$ preserves the lattice, i.e. $\frac{2}{r^2}*r*Q$ is an integer matrix.
+ - the pair $(r, v_0)$ is positive oriented, i.e. $(r, v_0) > 0$
+ - the product $(r, \~{r}) \geq \ 0$ for all roots $\~{r}$ already found
+ This implies that $r^2$ divides $2*i$ for $i$ being the level of $Q$, i.e. the last invariant of the smith normal form of $Q$. 
+
+ $P$ can be constructed by solving $(r, v_0) = n$ and $r^2 = k$ by increasing order of the value $\frac{n^2}{k}$ and $r$ satisfying the above conditions.
+
+ Perhaps $v_0$ lies on one or on several roots. Then $P$ is cannot be uniquely determined.
```

Comment:
Grammar:
```suggestion
 Perhaps $v_0$ lies on one or on several roots. Then $P$ is not uniquely determined.
```
But I would also say this sentence is overall a bit awkward. How about this:
```suggestion
 If $v_0$ lies on multiple roots then $P$ is not uniquely determined.
```